Libraries Tasmania.js

```diff
@@ -0,0 +1,781 @@
+			for (let row of doc.querySelectorAll("div#marc-tab0 table tr")) {
+				let fields = row.querySelectorAll("td");
+				fieldTag = ZU.trim(fields[0].textContent);
+				indicators = ZU.trim(fields[1].textContent);
+				// There don't seem to be any subfields in the records.
+				// Or at least I haven't found any.
+				// Adding in a subfield is necessary to get the translator to work.
+				fieldContent = marc.subfieldDelimiter + "a" + ZU.trim(fields[2].textContent);
+				record.addField(fieldTag, indicators, fieldContent);
+			}
+			record.translate(item);
+			// Doesn't seem to get format correctly from MARC, so add manually
+			item.itemType = format;
+			// Add reference number
+			item.callNumber = getFieldText(doc, "DOC_ID");
+			// Add permalink
+			item = addPermalink(doc, item, null);
+			// Add digitised files as attachments
+			addDigitalFiles(doc, item);
+		});
+	// No MARC so we have to scrape manually
+	}
+	else {
+		var item = new Zotero.Item(format);
+		item.title = getFieldText(doc, "T245_DISPLAY");
+		item.callNumber = getFieldText(doc, "DOC_ID");
+
+		// Creators
+		let authors = getLinkLists(doc, "AUTHOR_TALIS_FULL", "AUTHOR_INDEX");
+		for (let i = 0; i < authors.length; ++i) {
+			item.creators.push(ZU.cleanAuthor(authors[i], "author", true));
+		}
+		let contribs = getLinkLists(doc, "AUTHOR_OTHER_TALIS_FULL", "AUTHOR_INDEX");
+		for (let i = 0; i < contribs.length; ++i) {
+			item.creators.push(ZU.cleanAuthor(contribs[i], "contributor", true));
+		}
+
+		// PUBLICATION DETAILS
+		var pubInfo = getFieldText(doc, "PUBLICATION_INFO");
+		// Music can have a distributor
+		if (!pubInfo) {
+			pubInfo = getFieldText(doc, "DISTRIBUTION_264");
+		}
+		// eBooks?
+		if (!pubInfo) {
+			pubInfo = getFieldText(doc, "PUBLICATION_264");
+		}
+		// See if it matches standard format
+		var pubParts = pubInfo.match(/^(.+):(.+),(.+)$/);
+		if (pubParts) {
+			item.place = cleanText(pubParts[1]);
+			item.publisher = cleanText(pubParts[2]);
+			item.date = cleanText(pubParts[3]);
+		}
+		else {
+			item.publisher = pubInfo;
+		}
+
+		let contents = getFieldText(doc, "CONTENTS_TALIS");
+		let summary = getFieldText(doc, "SUMMARY_TALIS");
+		let notes = getFieldText(doc, "NOTES_TALIS");
+		let combined = [summary, contents, notes].join("\n\n");
+		item.abstractNote = combined ? combined : "";
+		
+		// Physical description
+		let physDesc = getFieldText(doc, "DESC_TALIS");
+		let physParts = physDesc.split(";");
+		if (physParts.length == 2 && format == "book") {
+			item.numPages = cleanText(physParts[0]);
+			item["physical desription"] = cleanText(physParts[1]);
+		}
+		else {
+			item["physical desription"] = cleanText(physDesc);
+		}
+
+		item = addPermalink(doc, item, null);
+		addDigitalFiles(doc, item);
+	}
+}
+
+function scrapeArchives(doc) {
+	var item = new Zotero.Item("manuscript");
+
+	// Types should be Agency, Series, or Item
+	let typeLabel = ZU.trim(doc.querySelector(".T245_DISPLAY_label").textContent).replace(/:$/, "");
```

Comment:
Stuff like this should use `text(doc, ...)` instead of `querySelector(...).textContent` for null-safety:

```suggestion
	let typeLabel = ZU.trim(text(doc, ".T245_DISPLAY_label")).replace(/:$/, "");
```

Reply:
Where does `text()` come from?

Reply:
From here: https://github.com/zotero/translate/blob/e08e318997b80685f03fbc4aae2cf6944fe46cfc/src/translation/translate.js#L1879-L1896

It's added to the translation sandbox for web and search translators. There's also `innerText`, `attr` (same idea but takes an attribute name as the last argument and returns the value of that attribute on the matched element, or the empty string), and now `request` (async HTTP).

Reply:
Changed in this instance and in a couple of other places.